apps/docs/app/[...slug]/_page.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * [Charis 2024-04-24]
+ * Temporarily turned off because it breaks the navigation back from the
+ * reference pages. Can turn this back on after we have completed the entire
+ * App Router migration.
+ *
+ * A catch-all to redirect all not-found pages.
+ *
+ * This allows us to generate recommendations based on the path the user is
+ * looking for, which isn't possible with a regular 404 page (receives no
+ * params).
+ */
+
+import { redirect } from 'next/navigation'
+import { notFoundLink } from '~/features/recommendations/NotFound.utils'
+
+type Params = { slug: string[] }
+
+const CatchAllPage = async ({ params }: { params: Params }) => {
```

Comment:
I tried this on the preview but it responds with a 500 error page.

<img width="1473" alt="Screenshot 2024-07-01 at 11 07 03" src="https://github.com/supabase/supabase/assets/25671831/de8e6fb6-06f0-4a1c-aaf4-158df6722a29">

Reply:
And maybe worth keeping this part as a separate PR to reduce scope? @charislam